sympy/integrals/integrals.py

```diff
@@ -694,6 +694,14 @@ def eval_factored(f, x, a, b):
                         try:
                             evalued = Add(*others)._eval_interval(x, a, b)
                             evalued_pw = piecewise_fold(Add(*piecewises))._eval_interval(x, a, b)
+                            if isinstance(evalued_pw, Integral):
+                                antideriv = self._eval_integral(function, xab[0], manual=True)
+                                piecewises = []
+                                for f in Add.make_args(antideriv):
+                                    if any(isinstance(g, Piecewise)
+                                           for g in Mul.make_args(f)):
+                                               piecewises.append(piecewise_fold(f))
```

Comment:
Here you get the pieces from the Piecewise?

Reply:
yes but i don't know whether this particular method is reasonable or not since I've reapeated some of the code again 
https://github.com/sympy/sympy/blob/c39a6848728a3c84a92f348a4fc5f1a4f4ad2b84/sympy/integrals/integrals.py#L682-L705

Reply:
Yes But i'm not sure of this approach